src/rocksdb.rs

```diff
@@ -508,6 +518,117 @@ impl DB {
         self.get_cf_opt(cf, key, &ReadOptions::default())
     }
 
+    pub fn multi_get(&self,
+                  keys: Vec<&[u8]>)
+                  -> Vec<Result<Option<DBVector>, String>> {
+        self.multi_get_opt(keys, &ReadOptions::default())
+    }
+
+    pub fn multi_get_opt(&self,
+                      keys: Vec<&[u8]>,
+                      readopts: &ReadOptions)
+                      -> Vec<Result<Option<DBVector>, String>> {
+
+        unsafe {
+            let mut err_ptrs : Vec<*mut i8> = vec![0 as *mut i8; keys.len()];
+            let mut value_ptrs : Vec<*mut u8> = vec![0 as *mut u8; keys.len()];
+            let mut length_ptrs : Vec<size_t> = vec![0 as size_t; keys.len()];
+
+            let mut key_lens : Vec<size_t> =  Vec::with_capacity(keys.len());
+            let mut key_ptrs : Vec<*const u8> = Vec::with_capacity(keys.len());
+            for ref key in &keys {
+                key_lens.push(key.len());
+                key_ptrs.push(key.as_ptr());
+            }
+
+            rocksdb_ffi::rocksdb_multi_get(self.inner,
+                                            readopts.inner,
+                                            keys.len() as size_t,
+                                            key_ptrs.as_ptr(),
+                                            key_lens.as_ptr(),
+                                            value_ptrs.as_mut_ptr(),
+                                            length_ptrs.as_mut_ptr(),
+                                            err_ptrs.as_mut_ptr());
+
+
+            let mut values = vec![];
+            for i in 0..keys.len() {
+
+                let err = err_ptrs[i];
+                if !err.is_null() {
+                    let em = error_message(err);
+                    values.push(Err(em));
+                }
+
+                let val = value_ptrs[i];
+                if val.is_null() {
+                    values.push(Ok(None));
+                } else {
+                    values.push(Ok(Some(DBVector::from_c(val, length_ptrs[i]))));
+                }
```

Comment:
could also continue the for loop in the error check if statement, or whatever you prefer to maintain 1:1 return values to requested keys.